src/main/java/org/jabref/gui/entryeditor/citationrelationtab/CitationRelationsTab.java

```diff
@@ -408,7 +414,31 @@ private void importEntries(List<CitationRelationItem> entriesToImport, CitationF
                 stateManager,
                 dialogService,
                 Globals.TASK_EXECUTOR);
-        importHandler.importEntries(entries);
+
+        if (searchType == CitationFetcher.SearchType.CITES) {
+            CitationKeyGenerator generator = new CitationKeyGenerator(databaseContext, preferencesService.getCitationKeyPatternPreferences());
+
+            List<String> citeKeys = new ArrayList<>();
+            for (BibEntry entryToCite : entries) {
+                if (entryToCite.getCitationKey().isEmpty()) {
+                    String key = generator.generateKey(entryToCite);
+                    entryToCite.setCitationKey(key);
+                    citeKeys.add(key);
+                }
+            }
+            existingEntry.setField(StandardField.CITES, citeKeys.stream().collect(Collectors.joining(",")));
+            importHandler.importEntries(entries);
+        }
+        if (searchType == CitationFetcher.SearchType.CITED_BY) {
+            for (BibEntry entryThatCitesOurExistingEntry : entries) {
+                List<String> existingCites = Arrays.stream(entryThatCitesOurExistingEntry.getField(StandardField.CITES).orElse("").split(",")).collect(Collectors.toList());
+                existingCites.removeIf(String::isEmpty);
+                existingCites.add(existingEntry.getCitationKey().orElse(""));
+                entryThatCitesOurExistingEntry.setField(StandardField.CITES, existingCites.stream().collect(Collectors.joining(",")));
+            }
+
+            importHandler.importEntries(entries);
+        }
```

Comment:
I think this should only do something, if the automatic key generation is turned on in the prefs, otherwise complain with a dialog?

Reply:
I like the "magic behavior" as we cannot add a relation if there is no key.

However, the preferences should be respected: If there is a key **existing** and we should generate keys and **overwrite**, we should overwrite.

Reply:
Currently the overwriting is done at the import, e.g. respecing the preferences. But I agree that it could be too late and we need to do the overwrite here as well if enabled